src/core/type.ml

```diff
@@ -1862,6 +1862,12 @@ let rec type_eq param a b =
 		List.iter2 (type_eq param) tl1 tl2
 	| TAnon a1, TAnon a2 ->
 		(try
+			(match !(a2.a_status) with
+			| Statics c -> (match !(a1.a_status) with Statics c2 when c == c2 -> () | _ -> error [])
+			| EnumStatics e -> (match !(a1.a_status) with EnumStatics e2 when e == e2 -> () | _ -> error [])
+			| AbstractStatics a -> (match !(a1.a_status) with AbstractStatics a2 when a == a2 -> () | _ -> error [])
+			| _ -> ()
```

Comment:
This is the same check used in `unify_anon` but I had to remove `| Opened -> a2.a_status := Closed` for the tests to compile – I'm not sure if this is correct or not!

https://github.com/HaxeFoundation/haxe/blob/7032594c6bd3d84f6abcf3bfcb3448a8389a19de/src/core/type.ml#L2231